backend/src/main/java/storage/repository/GenericRepository.java

```diff
@@ -0,0 +1,240 @@
+package storage.repository;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import model.DataModel;
+import storage.DBInterface;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Basic repository class that can take any model.
+ *
+ * Several things are presumed:
+ * - The model must have a default constructor
+ * - The model must set default values
+ * - The model must only use int, boolean or string (currently)
+ */
+public class GenericRepository <T extends DataModel> {
+
+    private final Gson gson;
+    private final Class<T> type;
+    private DBInterface dbInterface;
+
+    public GenericRepository(Class<T> type, DBInterface dbInterface){
+        this.gson = new Gson();
+        this.dbInterface = dbInterface;
+        this.type = type;
+    }
+
+    /***
+     * gets all objects matching given type
+     * @return
+     */
+    public List<T> getObjects() {
+        String query = "SELECT * FROM "+ type.getSimpleName();
+        System.out.println("Running query: "+query);
+        ResultSet rs =  this.dbInterface.executeQuerry(query);
+        try {
+            return this.getMatchJSON(rs); //converts resulting set to model instance
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Gets specific instance of model
+     * @param id
+     * @return
+     */
+    public T getObject(int id) {
+        String query = "SELECT * FROM "+ type.getSimpleName() +" WHERE id="+id;
+        System.out.println("Running query: "+query);
+        ResultSet rs =  this.dbInterface.executeQuerry(query);
+        try {
+            return this.getMatchJSON(rs).iterator().next(); //converts resulting set to model instance
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Creates an given instance om model
+     * @param object
+     * @return
+     */
+    public T postObject(T object) {
+        JsonObject jsonObject = this.gson.toJsonTree(object).getAsJsonObject();
+        StringBuilder query = new StringBuilder("INSERT INTO "+ type.getSimpleName() +" (");
+
+        for (String key : jsonObject.keySet()){
+            if(Objects.equals(key, "id"))continue; //skip ID
+            String last = query.substring(query.length() - 1);
+            if(!last.equals("(")){ //first
+                query.append(", ");
+            }
+            query.append(key);
+        }
+        query.append(")");
+
+        for(int i = 0; i < jsonObject.keySet().size(); i++){
+            query.append(i==0?" VALUES (":i==1?"?":",?"); //skip ID
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)